biz.aQute.bndlib/src/aQute/bnd/classfile/Attribute.java

```diff
@@ -12,6 +12,54 @@
 
 @ProviderType
 public interface Attribute {
+
+	@ProviderType
+	enum AttributeTag {
+		Unrecognized(UnrecognizedAttribute.class), //
+		Preview(Attribute.class), //
+		AnnotationDefault(AnnotationDefaultAttribute.class), //
+		BootstraMethods(BootstrapMethodsAttribute.class), //
+		Code(CodeAttribute.class), //
+		ConstantValue(ConstantValueAttribute.class),
+		Deprecated(DeprecatedAttribute.class),
+		EnclosingMethod(EnclosingMethodAttribute.class),
+		Exceptions(ExceptionsAttribute.class),
+		InnerClasses(InnerClassesAttribute.class), //
+		LineNumberTable(LineNumberTableAttribute.class),
+		LocalVariableTable(LocalVariableTableAttribute.class), //
+		LocalVariableTypeTable(LocalVariableTypeTableAttribute.class), //
+		MethodParameters(MethodParametersAttribute.class), //
+		Module(ModuleAttribute.class), //
+		ModuleMainClass(ModuleMainClassAttribute.class),
+		ModulePackages(ModulePackagesAttribute.class), //
+		NestHost(NestHostAttribute.class), //
+		NestMembers(NestMembersAttribute.class), //
+		RuntimeInvisibleAnnotations(RuntimeInvisibleAnnotationsAttribute.class), //
+		RuntimeInvisibleParameterAnnotations(RuntimeInvisibleParameterAnnotationsAttribute.class), //
+		RuntimeInvisibleTypeAnnotations(RuntimeInvisibleTypeAnnotationsAttribute.class), //
+		RuntimeVisibleAnnotations(RuntimeVisibleAnnotationsAttribute.class), //
+		RuntimeVisibleParameterAnnotations(RuntimeVisibleParameterAnnotationsAttribute.class), //
+		RuntimeVisibleTypeAnnotations(RuntimeVisibleTypeAnnotationsAttribute.class), //
+		Signature(SignatureAttribute.class), //
+		SourceDebugExtension(SourceDebugExtensionAttribute.class), //
+		SourceFile(SourceFileAttribute.class), //
+		StackMapTable(StackMapTableAttribute.class), //
+		Synthetic(SyntheticAttribute.class);
+
+		public Class<? extends Attribute> clazz;
+
+		AttributeTag(Class<? extends Attribute> clazz) {
+			this.clazz = clazz;
+		}
+
+		@SuppressWarnings("unchecked")
+		public <T extends Attribute> T cast(Attribute attr) {
+			return (T) clazz.cast(attr);
```

Comment:
This is not really a safe case since T is not bound to anything. For example the compiler will accept 

```java
SignatureAttribute foo = AttributeTag.Preview.cast(new RecordAttribute(...));
```
So it is a lot of work to get something not very safe.

If you want to make an easy (but perhaps not perfectly safe cast), then just add something like

```java
<A extends Attribute> A cast();
```

to `Attribute` and then each attribute implementation will do something like:

```java
	@SuppressWarnings("unchecked")
	@Override
	public SignatureAttribute cast() {
		return this;
	}
```

Then if you have `Attribute attribute`, you can do `SignatureAttribute sigAttr = attribute.cast();`